streams/dshr_strdata_mod.F90

```diff
@@ -2031,61 +2094,95 @@ subroutine shr_strdata_set_stream_iodesc(sdat, per_stream, fldname, pioid, rc)
     rcode = pio_inq_vartype(pioid, varid, pio_iovartype)
 
     ! determine io descriptor
+    !-------------------------------
     if (ndims == 2) then
+    !-------------------------------
        rcode = pio_inq_dimname(pioid, dimids(ndims), dimname)
-       if (trim(dimname) == 'time' .or. trim(dimname) == 'nt') then
-          if (sdat%mainproc) then
-             write(sdat%stream(1)%logunit,F03) 'setting iodesc for : '//trim(fldname)// &
-                  ' with dimlens(1) = ',dimlens(1),' and the variable has a time dimension '
+       if ((trim(dimname) == 'time' .or. trim(dimname) == 'nt')) then
+          if (mainproc) then
+             write(logout,'(2a,2(i0,2x),a)') trim(subname),' setting iodesc for 2d: '//trim(fldname)// &
```

Comment:
Does this work? I only see one integer:
```suggestion
             write(logout,'(2a,i0,2x,a)') trim(subname),' setting iodesc for 2d: '//trim(fldname)// &
```
I also think it would be easier to check (and more efficient time wise) if you used separate arguments instead of a concatenation (applies to whole module).

Reply:
done